### DISCUSSION TITLE
How to config custom key-value pairs in mkdocs.yml ?

### DISCUSSION THREAD

**Discussion by wanzhenchn** [squidfunk/mkdocs-material#5084] [Q&A] [answered]
I want to add NEW key-value pairs (like pdf_url) in mkdocs.yml, for example,

repo_url: http://xxxxxx
**pdf_url**: http://xxxxxx

however, the

`mkdocs build --strict`

shows that WARNING  -  Config value 'pdf_url': Unrecognised configuration name: pdf_url.

So how to config custom key-value pairs in mkdocs.yml ?

**Comment by kamilkrzyskow** [comment, marked as answer]
The `repo_url` is a predefined setting of the MkDocs configuration file. As far as I know, you cannot add more global key value pairs there without changing how it works. However, there is the `extra` key, which allows to add any keys under there.
Like so:
```yml
extra:
    pdf_url: http://xxxxxx/
```
and then in your Jinja2 HTML template you can do something like this:
```jinja2
<a href="{{ config.extra.pdf_url }}">Download the PDF</a>
```
or maybe you want to have access to those variables in the Markdown, then maybe the `macros` plugin can help you.

https://www.mkdocs.org/user-guide/configuration/#extra
https://mkdocs-macros-plugin.readthedocs.io/en/latest/

**Reply by wanzhenchn** [reply]
> 

Right, many thanks.
